source/welcome.svelte

```diff
@@ -0,0 +1,132 @@
+<svelte:options customElement='rgh-welcome' />
+
+<script lang='ts'>
+	import './welcome.css';
+	import {onMount} from 'svelte';
+
+	import './helpers/target-blank-polyfill.js';
+	import optionsStorage from './options-storage.js';
+	import {baseApiFetch} from './github-helpers/github-token.js';
+
+	let stepVisible = 1;
+	let stepValid = 0;
+	let tokenInput = '';
+	let tokenError = '';
+
+	$: if (stepValid === 1) {
+		setTimeout(showThirdStep, 2000);
+	}
+
+	$: if (stepValid === 3) {
+		setTimeout(() => {
+			location.href = 'https://github.com/refined-github/refined-github/wiki';
+		}, 2000);
+	}
+
+	$: if (tokenInput) {
+		verifyToken();
+
+		// @ts-expect-error TS and its index signatures...
+		optionsStorage.set({personalToken: tokenInput});
+	}
+
+	const origins = ['https://github.com/*', 'https://gist.github.com/*'];
+
+	async function grantPermissions() {
+		const granted = await chrome.permissions.request({origins});
+		if (granted) {
+			stepVisible = 2;
+			stepValid = 1;
+		}
+	}
+
+	function showThirdStep() {
+		stepVisible = 3;
+	}
+
+	function markSecondStep() {
+		setTimeout(() => {
+			stepValid = 2;
+			stepVisible = 3;
+		}, 1000);
+	}
+
+	async function verifyToken() {
+		try {
+			await baseApiFetch({apiBase: 'https://api.github.com/', token: tokenInput, path: '/'});
+			stepValid = 3;
+			tokenError = '';
+		} catch {
+			tokenError = 'Invalid token';
+		}
+	}
+
+	onMount(async () => {
+		if (await chrome.permissions.contains({origins})) {
+			stepValid = 1;
+			setTimeout(() => {
+				stepVisible = 2;
+			}, 500);
+		}
+	});
+</script>
+
+<link rel='stylesheet' href='welcome.css'>
+<main class:dimmed={stepValid === 3}>
+	<header>
+		<h1>
+			<img src='icon.png' alt="" height='32'>
+			Welcome to Refined GitHub
+		</h1>
+	</header>
+	<ul>
+		<li class:valid={stepValid >= 1} class:visible={stepVisible >= 1} class='will-show'>
+			{#if stepValid === 0}
+				<button on:click={grantPermissions}>
+					Grant
+				</button>
+			{:else}
+				Grant
+			{/if}
+			the extension access to github.com
+		</li>
+
+		<!-- svelte-ignore a11y-no-noninteractive-element-interactions -->
+		<!-- svelte-ignore a11y-click-events-have-key-events -->
+		<li class:valid={stepValid >= 2} class:visible={stepVisible >= 2} class='will-show' on:click={showThirdStep}>
+			<a
+				href='https://github.com/settings/tokens/new?description=Refined%20GitHub&scopes=repo,read:project'
```

Comment:
@djhoese thanks for the comment!

The link should already pre-select the minimal scopes we expect, see the link here. Does it not happen for you? What's your browser? This is what you should see:

![IMG_6045](https://github.com/user-attachments/assets/7898275b-daa1-472f-8970-136b6931372f)

Reply:
Ah, that would do it. I didn't click that link. I went to the token page myself. Makes sense.

Reply:
I'll see if I can add that information in the wiki as well though, and/or I'll bring out the same UI we have in the options page.